application/models/SurveysGroups.php

```diff
@@ -323,22 +326,22 @@ public function getButtons()
 
     /**
      * Get the group list for current user
+     * @gsid null|integer : allow to restrict to a specigf gif
      * @return array
      */
-    public static function getSurveyGroupsList()
+    public static function getSurveyGroupsList($gsid = null)
```

Comment:
I am not sure about this.
I understand the usefullness.

Still, name wouldn't match the behaviour when using it with an argument.
Doesn't feel right.

If needed, maye we should:
a) create a new method? What name would it be?
b) create an alias to expose the getPermissionCriteria method?

Right now sounds like something which may need to be unwinded and the more we use it, the hardest will be to unwind it

Reply:
Both can be done, but for getPermissionCriteria i think it's more related to https://bugs.limesurvey.org/view.php?id=16459 (and moving it to Trait)

Reply:
We create an alias now, to expose it.
If later we move it to a trait, it will still work. right?

Reply:
Moving to scope

Reply:
>  If later we move it to a trait, it will still work. right?

See https://github.com/LimeSurvey/LimeSurvey/pull/2754